processor/processor.go

```diff
@@ -535,6 +509,39 @@ func (proc *HandleT) syncTransformerFeatureJson(ctx context.Context) {
 	}
 }
 
+func (proc *HandleT) makeFeaturesFetchCall() bool {
+	url := transformerURL + "/features"
+	req, err := http.NewRequest("GET", url, bytes.NewReader([]byte{}))
+	if err != nil {
+		proc.logger.Error("error creating request - %s", err)
+		return true
+	}
+	tr := &http.Transport{}
+	client := &http.Client{Transport: tr}
+	res, err := client.Do(req)
+	if err != nil {
+		proc.logger.Error("error sending request - %s", err)
+		return true
+	}
+
+	defer res.Body.Close()
+
+	if res.StatusCode == 200 {
+		body, err := io.ReadAll(res.Body)
+		if err == nil {
+			proc.transformerFeatures = json.RawMessage(body)
+			return false
+		}
+
+		return true
+	} else if res.StatusCode == 404 {
+		proc.transformerFeatures = json.RawMessage(defaultTransformerFeatures)
+		return false
+	}
+
```

Comment:
Body is not read if `res.StatusCode != 200`
```suggestion
	defer res.Body.Close()
	body, err := io.ReadAll(res.Body)
	if err != nil {
	     // error logging here:
	     return true
	}
	if res.StatusCode == 200 {
			proc.transformerFeatures = json.RawMessage(body)
			return false
	} else if res.StatusCode == 404 {
		proc.transformerFeatures = json.RawMessage(defaultTransformerFeatures)
		return false
	}

```

Reply:
done